res/controllers/Numark_DJ2GO2_Touch_scripts.js

```diff
@@ -0,0 +1,192 @@
+            this.previewStarted = true;
+            this.previewSeekEnabled = true;
+        }
+    },
+    input: function(channel, control, value, status, _group) {
+        switch (status) {
+        case 0xBF: // Rotate.
+            var rotateValue = (value === 127) ? -1 : ((value === 1) ? 1 : 0);
+            this.onKnobEvent(rotateValue);
+            break;
+        case 0x9F: // Push.
+            this.onButtonEvent();
+        }
+    }
+});
+
+DJ2GO2Touch.masterGain = new components.Pot({
+    midi: [0xBF, 0x0A],
+    group: "[Master]",
+    key: "gain"
+});
+
+DJ2GO2Touch.cueGain = new components.Pot({
+    midi: [0xBF, 0x0C],
+    group: "[Master]",
+    key: "headGain"
+});
+
+DJ2GO2Touch.crossfader = new components.Pot({
+    midi: [0xBF, 0x08],
+    group: "[Master]",
+    key: "crossfader"
+});
+
+// implement a constructor for a custom Deck object specific to your controller
+DJ2GO2Touch.Deck = function(deckNumbers, midiChannel) {
+    // Call the generic Deck constructor to setup the currentDeck and deckNumbers properties,
+    // using Function.prototype.call to assign the custom Deck being constructed
+    // to 'this' in the context of the generic components.Deck constructor
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call
+    components.Deck.call(this, deckNumbers);
+
+    this.playButton = new components.PlayButton([0x90 + midiChannel, 0x00]);
+    this.cueButton = new components.CueButton([0x90 + midiChannel, 0x01]);
+    this.syncButton = new components.SyncButton([0x90 + midiChannel, 0x02]);
+
+    this.pflButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x1B],
+        key: "pfl"
+    });
+
+    this.loadButton = new components.Button({
+        midi: [0x9F, 0x02 + midiChannel],
+        key: "LoadSelectedTrack",
+        input: function(channel, control, value, status, _group) {
+            this.send(this.isPress(channel, control, value, status) ? this.on : this.off);
+            components.Button.prototype.input.apply(this, arguments);
+        }
+    });
+
+    this.preGain = new components.Pot({
+        midi: [0xB0 + midiChannel, 0x16],
+        group: "[QuickEffectRack1_" + this.currentDeck + "]",
+        key: "super1"
+    });
+
+    engine.setValue(this.currentDeck, "rate_dir", -1);
+    this.tempoFader = new components.Pot({
+        group: "[Channel" + deckNumbers + "]",
+        midi: [0xB0 + midiChannel, 0x09],
+        key: "rate",
+        inSetParameter: components.Pot.prototype.inSetParameter,
+        connect: function() {
+            engine.softTakeover(this.group, "rate", true);
+            components.Pot.prototype.connect.apply(this, arguments);
+        }
+    });
+
+    this.hotcueButtons = [];
+    this.samplerButtons = [];
+    this.beatloopButtons = [];
+    for (var i = 1; i <= 4; i++) {
+        this.hotcueButtons[i] = new components.HotcueButton({
+            group: "[Channel" + deckNumbers + "]",
```

Comment:
this is very brittle and only works because `deckNumbers` is an array containing just a single number. 
parsing the group using `script.deckFromGroup(this.currentDeck)` is probably more robust.